tests/Monolog/HandlerTest.php

```diff
@@ -281,5 +281,30 @@ public function handleDataProvider(): \Generator
                 'bar.tag' => 'bar tag value',
             ],
         ];
+
+        yield [
+            [
+                'message' => 'foo bar',
+                'level' => Logger::INFO,
+                'level_name' => Logger::getLevelName(Logger::INFO),
+                'channel' => 'channel.foo',
+                'context' => [
+                    'extra' => [
+                        1 => 'numeric key',
+                    ],
+                ],
+                'extra' => [],
+            ],
+            [
+                'level' => Severity::info(),
+                'message' => 'foo bar',
+            ],
+            [
+                'monolog.channel' => 'channel.foo',
+                'monolog.level' => Logger::getLevelName(Logger::INFO),
+                1 => 'numeric key',
```

Comment:
I would expect this to be a `string` and not an `integer`. Test is passing because the wonderful PHP casts strings containing numbers without decimals to integers when they are used as key of the arrays and so the comparison succeeds :man_facepalming:

```php
$a = [
    1 => 'foo',
];

$b = [
    '1' => 'foo',
];

var_dump($a == $b); // true
var_dump($a === $b); // true

$a = [
    1.2 => 'foo',
];

$b = [
    '1.2' => 'foo',
];

var_dump($a == $b); // false
var_dump($a === $b); // false
```

Please change the test case so that a float is used, this way the test should fail if something is wrong in the comparison of the expected output with the input data

Reply:
Attempting to use floats as array keys causes them to get truncated to integers, so that won't help.
```php
$a = [
    1.2 => 'foo',
];

$b = [
    '1' => 'foo',
];

var_dump($a == $b); // true
var_dump($a === $b); // true
```
The test doesn't pass without the cast (due to the exception) though, so from that point of view it's functional.

Where in the changelog should I add the notice, under 2.1.1?

Reply:
>Where in the changelog should I add the notice, under 2.1.1?

Under the `Unreleased` section :+1:

Reply:
Sounds good!